### ISSUE TITLE
Frozen app forever

### ISSUE THREAD

**Issue by Kryptak** [RenegadeLucien/project-tenacity#14]
I followed instructions but i couldn't make it work.

1- Latest java 8.
2- open app / create profile / my RS display name
3- app created ge txt file
4- app freezes, cpu load and network activity around few minutes.
5- cpu load, no network activity, app still frozen and ge txt file empty.
6- after 1 hour nothing changed.

I tried it couple of times but no luck.

**Comment by RenegadeLucien** [comment]
Can you try to run the jar from the command line? (on Windows, open "Command Prompt" from the start menu, then use the "java -jar (location of Project Tenacity jar)" command)
This will provide lots of debug logging that might help narrow down this problem. Additionally, if any errors occur, it should log an error message.

**Comment by Kryptak** [comment]
java.io.IOException: Server returned HTTP response code: 403 for URL: https://runescape.wiki/w/Module:Exchange/Black_salamander?action=raw
        at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(Unknown Source)
        at sun.net.www.protocol.http.HttpURLConnection.getInputStream(Unknown Source)
        at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(Unknown Source)
        at java.net.URL.openStream(Unknown Source)
        at data.dataobjects.Item.getData(Item.java:64)
        at data.dataobjects.Item.<init>(Item.java:27)
        at data.databases.ItemDatabase.addItems(ItemDatabase.java:239)
        at data.databases.ItemDatabase.<init>(ItemDatabase.java:17)
        at data.databases.ItemDatabase.getItemDatabase(ItemDatabase.java:1768)
        at data.dataobjects.Achievement.getTimeForRequirements(Achievement.java:268)
        at logic.Player.calcAllAchievements(Player.java:349)
        at ui.Planner.displayTasks(Planner.java:87)
        at ui.Planner.lambda$profileCreation$8(Planner.java:752)
        at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
        at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
        at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
        at com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
        at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
        at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
        at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:49)
        at javafx.event.Event.fireEvent(Event.java:198)
        at javafx.scene.Node.fireEvent(Node.java:8411)
        at javafx.scene.control.Button.fire(Button.java:185)
        at com.sun.javafx.scene.control.behavior.ButtonBehavior.mouseReleased(ButtonBehavior.java:182)
        at com.sun.javafx.scene.control.skin.BehaviorSkinBase$1.handle(BehaviorSkinBase.java:96)
        at com.sun.javafx.scene.control.skin.BehaviorSkinBase$1.handle(BehaviorSkinBase.java:89)
        at com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:218)
        at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
        at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
        at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
        at com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
        at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
        at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
        at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
        at javafx.event.Event.fireEvent(Event.java:198)
        at javafx.scene.Scene$MouseHandler.process(Scene.java:3757)
        at javafx.scene.Scene$MouseHandler.access$1500(Scene.java:3485)
        at javafx.scene.Scene.impl_processMouseEvent(Scene.java:1762)
        at javafx.scene.Scene$ScenePeerListener.mouseEvent(Scene.java:2494)
        at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:394)
        at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:295)
        at java.security.AccessController.doPrivileged(Native Method)
        at com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$358(GlassViewEventHandler.java:432)
        at com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:389)
        at com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(GlassViewEventHandler.java:431)
        at com.sun.glass.ui.View.handleMouseEvent(View.java:555)
        at com.sun.glass.ui.View.notifyMouse(View.java:937)
        at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
        at com.sun.glass.ui.win.WinApplication.lambda$null$152(WinApplication.java:177)
        at java.lang.Thread.run(Unknown Source)

**Comment by RenegadeLucien** [comment]
Alright, so I can't reproduce this locally so I can't be sure that this is fixed. But it looks an awful lot like the issue described in [this StackOverflow post.](https://stackoverflow.com/questions/13670692/403-forbidden-with-java-but-not-web-browser) I added a patch in accordance with the responses to this post that should hopefully resolve this issue.

The recently-released v0.8.3.1b contains this patch, let me know if it works now.

**Comment by Kryptak** [comment]
> Alright, so I can't reproduce this locally so I can't be sure that this is fixed. But it looks an awful lot like the issue described in [this StackOverflow post.](https://stackoverflow.com/questions/13670692/403-forbidden-with-java-but-not-web-browser) I added a patch in accordance with the responses to this post that should hopefully resolve this issue.
> 
> The recently-released v0.8.3.1b contains this patch, let me know if it works now.

i opened it via command line and it works, loaded GE items, quests, tasks etc.

it is now on:
```
Steel, Stud, or Silk Salute     6       13.3314322
New Varrock Set Tasks - Elite   18      41.178565
South AND West Side Story       6       13.467999
Pick Half       0       5.643E-4
Pick All        2       4.5184086
New Varrock Set Tasks - Hard    14      31.2937956
Tier Gear       0       3.633E-4
They Had a Gudrun       6       13.5338334
Scrying Game    2       4.4863971
New Varrock Set Tasks - Medium
```
little bit slower but it seems like its gonna work, good job and tyvm!

**Comment by RenegadeLucien** [comment]
Awesome, glad it's working =)
